From 22c13ee169ef0a781c1f11111bfda165d842abdf Mon Sep 17 00:00:00 2001
From: Timon Sassor <timon.sassor@governikus.de>
Date: Tue, 19 Aug 2025 18:12:12 +0200
Subject: Android: Set name-matched class names for Slider and ScrollBar for
 TalkBack

Seekbar is not the ideal class name for ScrollBar and Slider. It changes
the behavior of these Element, so that the scroll actions are not
properly delivered to them by TalkBack. We use the Android class name
that best match the QAccessible role name. We accept, that TalkBack has
no implementation for them and will use the default behavior of
Android.view.

[ChangeLog][Accessibility][Android] Use different Android class names
for Slider and Scroller.

Pick-to: 6.10 6.9 6.8
Change-Id: I03d9ce2bd051ff2c330852636d92e47c6d971884
Reviewed-by: Assam Boudjelthia <assam.boudjelthia@qt.io>
(cherry picked from commit b682d31d7e58cd74bb32df2fc984325f0825020e)
---
 src/plugins/platforms/android/androidjniaccessibility.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git x/qtbase/src/plugins/platforms/android/androidjniaccessibility.cpp y/qtbase/src/plugins/platforms/android/androidjniaccessibility.cpp
index 4795fffe3143f33e5fc2e142cb67cda55de159c4..64a1065f32121e1cc7466a9b7ec07cc1c8f6a1f4 100644
--- x/qtbase/src/plugins/platforms/android/androidjniaccessibility.cpp
+++ y/qtbase/src/plugins/platforms/android/androidjniaccessibility.cpp
@@ -485,9 +485,10 @@ namespace QtAndroidAccessibility
             return QStringLiteral("android.app.ActionBar.Tab");
         case QAccessible::Role::PageTabList:
             return QStringLiteral("android.widget.TabWidget");
-        case QAccessible::Role::ScrollBar: [[fallthrough]];
+        case QAccessible::Role::ScrollBar:
+            return QStringLiteral("android.widget.Scroller");
         case QAccessible::Role::Slider:
-            return QStringLiteral("android.widget.SeekBar");
+            return QStringLiteral("com.google.android.material.slider.Slider");
         case QAccessible::Role::Table:
             // #TODO Evaluate the usage of AccessibleNodeInfo.setCollectionItemInfo() to provide
             // infos about colums, rows und items.
